Lib/fontTools/otlLib/optimize.py

```diff
@@ -468,8 +468,7 @@ def make_cluster(indices: int) -> Cluster:
             # See https://github.com/harfbuzz/packtab/blob/master/Lib/packTab/__init__.py#L690-L691
             # Given the size reduction we have achieved so far, compute how many
             # new subtables are acceptable.
-            size_reduction_per_subtable = 1 - 0.5 ** (1 / compression)
-            max_new_subtables = size_reduction / size_reduction_per_subtable
+            max_new_subtables = size_reduction * 2 * compression
```

Comment:
I believe both are wrong in implementing the criteria I had in mind.
```patch
diff --git a/Lib/fontTools/otlLib/optimize.py b/Lib/fontTools/otlLib/optimize.py
index 70390182..a5ec06a1 100644
--- a/Lib/fontTools/otlLib/optimize.py
+++ b/Lib/fontTools/otlLib/optimize.py
@@ -2,6 +2,7 @@ from collections import defaultdict
 import enum
 from functools import reduce
 from itertools import chain
+from math import log2
 from sys import stderr
 from typing import Any, DefaultDict, Dict, Iterable, List, Sequence, Set, Tuple
 
@@ -465,7 +466,7 @@ def cluster_pairs_by_class2_coverage_custom_cost(
             # See https://github.com/harfbuzz/packtab/blob/master/Lib/packTab/__init__.py#L690-L691
             # Given the size reduction we have achieved so far, compute how many
             # new subtables are acceptable.
-            max_new_subtables = size_reduction * 2 * compression
+            max_new_subtables = -log2(1 - size_reduction) * compression
             print(
                 f"            len(clusters) = {len(clusters):3d}    size_reduction={size_reduction:5.2f}    max_new_subtables={max_new_subtables}",
                 file=stderr,
```

Doesn't make much of a difference in the one example I tried.

I suppose you did most of your analysis using `size_reduction` of .5, which mades all three of these equivalent...

Consider `size_reduction` of .75 and the differences should become more obvious.

Reply:
Thinking more, I think something linear like what you had is more adequate here than the log one I had in packtab. Again, it shouldn't matter. But a pathological case is if the matrix was pretty sparse, say, 100x100 matrix with only 100 non-zero entries... are what would we want at any compression level? A hundred subtables is not the best idea...

In reality, that should have been made a PairPosFormat1 perhaps....

Reply:
I'll think more about it. Am falling towards maybe just using `max_new_subtables = compression`. But that would push a lot of fonts towards 5 subtables with negligible filesize savings. That's exactly why I did the packtab tradeoff.

Let me think more about it. Maybe comparing to size of the fully-clustered is the wrong approach... Humm. I've confused myself.

Reply:
Umm. Thinking more, let's keep the log function.

Reply:
Thanks. What is your reasoning behind the log? So that I can write it in a comment next to the formula in the code